mobile/library/cc/engine_builder.cc

```diff
@@ -752,9 +752,13 @@ std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> EngineBuilder::generate
       *(*envoy_layer.mutable_fields())["envoy"].mutable_struct_value();
   ProtobufWkt::Struct& reloadable_features =
       *(*runtime_values.mutable_fields())["reloadable_features"].mutable_struct_value();
-  for (auto& guard_and_value : runtime_guards_) {
-    (*reloadable_features.mutable_fields())[guard_and_value.first].set_bool_value(
-        guard_and_value.second);
+  for (const auto& [name, is_enabled] : runtime_guards_) {
+    if (name == "dns_cache_set_ip_version_to_remove" && is_enabled &&
+        !dns_preresolve_hostnames_.empty()) {
```

Comment:
I think we want to check this where we actually use dns_preresolve_hostnames (and ideally reject config) if that's not problematic.

Reply:
Done. Updated the code to move the validation logic into `dns_cache_impl.cc` instead of EM.